File: fixtures/scale/julia/small/src/Banner.jl
module Banner

using ..Colors
using ..Align
using ..Border
using ..Log
using ..Config

export cmd_banner

function cmd_banner(text::String)
    width = Config.config_get("width")
    Log.log_info("banner width =", width)
    centered = Align.center_text(text, width)
    for line in Border.border_wrap([centered], width)
        println(line)
    end
end

function banner_preview(text::String)
    return Colors.colorize("cyan", Align.center_text(text, 40))
end

end

File: fixtures/scale/r/small/commands/banner.r
# banner subcommand: one centered, framed headline.

source("lib/colors.r")
source("lib/align.r")
source("lib/border.r")
source("lib/log.r")

cmd_banner <- function(text) {
  width <- config_get("width")
  log_info("banner width =", width)
  centered <- center_text(text, width)
  cat(border_wrap(centered, width), sep = "\n")
}

banner_preview <- function(text) {
  colorize("cyan", center_text(text, 40))
}
